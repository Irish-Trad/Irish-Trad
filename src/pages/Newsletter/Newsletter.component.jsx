import React from 'react'

import { MainContainer } from './Newsletter.styles'

const Newsletter = () => <MainContainer><h1>Newsletter</h1></MainContainer>

export default Newsletter