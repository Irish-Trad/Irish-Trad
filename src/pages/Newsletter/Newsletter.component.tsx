import React from 'react'
import Metadata from 'common/Metadata'

import {
  MainContainer,
  H1,
  Text,
  ExternalLink
} from 'common/StyledComponents/StyledComponents'

import {
  StyledLink,
  StyledIcon,
  ButtonContainer,
  Button
} from './Newsletter.styles'

const Newsletter: React.FunctionComponent = () => (
  <MainContainer>
    <Metadata
      title="Irish-Trad.net - Newsletter"
      metaTitle="Irish-Trad.net - Newsletter"
      metaKeywords="Irish Trad, Irish Traditional, Irish Tenor Banjo, Irish Bouzouki, Bouzouki, Mandolin, Octave Mandolin, Mandola, Octave Mandola, Chords, TAB, Tablature"
      metaDescription="Newsletter for Irish-Trad.net"
    />
    <H1>Newsletter</H1>
    <Text>
      Irish-Trad.net is a hobby project so progress is slow when developing new
      resources or lessons.
    </Text>
    <Text>
      If you would like to sign-up to receive an email when large changes have
      been made I have created a MailChimp subscription which can be found by
      clicking the following Icon or Button:
    </Text>
    <ButtonContainer>
      <StyledLink
        href="https://mailchi.mp/bfa9fa63aaab/irish-trad"
        target="_blank"
      >
        <StyledIcon icon={['fab', 'mailchimp']} />
      </StyledLink>
      <ExternalLink href="https://discord.gg/Mq9jeA8" target="_blank">
        <Button>Sign Up For Newsletter</Button>
      </ExternalLink>
    </ButtonContainer>
    <Text>
      MailChimp allows you to manage your subscription at anytime so
      unsubscribing is easy if you change your mind.
    </Text>
  </MainContainer>
)

export default Newsletter
