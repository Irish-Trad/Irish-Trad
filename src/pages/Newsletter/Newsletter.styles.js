import styled from 'styled-components'

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

export const StyledLink = styled.a``

export const StyledIcon = styled(FontAwesomeIcon)`
  font-size: 106px;
  color: #006400;
  margin: 10px;
`

export const ButtonContainer = styled.div`
  display: flex;
  flex-flow: row wrap;
  justify-content: space-evenly;
  align-items: center;
`

export const Button = styled.button`
  height: 50px;
  color: white;
  background: #006400;
  border: none;
  border-radius: 12px;
  font-size: 16px;
  padding: 10px;
  cursor: pointer;
`
