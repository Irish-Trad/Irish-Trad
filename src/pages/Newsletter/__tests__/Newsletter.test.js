import React from 'react'
import { render, cleanup } from '@testing-library/react'
import Newsletter from '../Newsletter.component'

describe('Newsletter', () => {
  afterEach(() => {
    cleanup()
  })

  it('Snapshot', () => {
    const snapshot = render(<Newsletter />)

    expect(snapshot).toMatchSnapshot()
  })

  it('Title of the page is Newsletter', () => {
    const { getByText } = render(<Newsletter />)

    expect(getByText('Newsletter')).toBeInTheDocument()
  })
})
