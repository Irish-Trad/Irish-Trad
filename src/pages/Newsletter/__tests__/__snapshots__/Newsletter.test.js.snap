// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Newsletter Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="sc-bdVaJa aBmTA"
      >
        <h1
          class="sc-bwzfXH kWcKFk"
        >
          Newsletter
        </h1>
        <p
          class="sc-bxivhb faZbih"
        >
          Irish-Trad.net is a hobby project so progress is slow when developing new resources or lessons.
        </p>
        <p
          class="sc-bxivhb faZbih"
        >
          If you would like to sign-up to receive an email when large changes have been made I have created a MailChimp subscription which can be found by clicking the following Icon or Button:
        </p>
        <div
          class="sc-gZMcBi hLSPcS"
        >
          <a
            class="sc-dnqmqq bDsesi"
            href="https://mailchi.mp/bfa9fa63aaab/irish-trad"
            target="_blank"
          >
            <i
              class="fa"
            />
          </a>
          <a
            class="sc-htoDjs gjSkhw"
            href="https://discord.gg/Mq9jeA8"
            target="_blank"
          >
            <button
              class="sc-gqjmRU isyvva"
            >
              Sign Up For Newsletter
            </button>
          </a>
        </div>
        <p
          class="sc-bxivhb faZbih"
        >
          MailChimp allows you to manage your subscription at anytime so unsubscribing is easy if you change your mind.
        </p>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="sc-bdVaJa aBmTA"
    >
      <h1
        class="sc-bwzfXH kWcKFk"
      >
        Newsletter
      </h1>
      <p
        class="sc-bxivhb faZbih"
      >
        Irish-Trad.net is a hobby project so progress is slow when developing new resources or lessons.
      </p>
      <p
        class="sc-bxivhb faZbih"
      >
        If you would like to sign-up to receive an email when large changes have been made I have created a MailChimp subscription which can be found by clicking the following Icon or Button:
      </p>
      <div
        class="sc-gZMcBi hLSPcS"
      >
        <a
          class="sc-dnqmqq bDsesi"
          href="https://mailchi.mp/bfa9fa63aaab/irish-trad"
          target="_blank"
        >
          <i
            class="fa"
          />
        </a>
        <a
          class="sc-htoDjs gjSkhw"
          href="https://discord.gg/Mq9jeA8"
          target="_blank"
        >
          <button
            class="sc-gqjmRU isyvva"
          >
            Sign Up For Newsletter
          </button>
        </a>
      </div>
      <p
        class="sc-bxivhb faZbih"
      >
        MailChimp allows you to manage your subscription at anytime so unsubscribing is easy if you change your mind.
      </p>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
